ci/bazel-scripts/diff.sh

```diff
@@ -3,24 +3,19 @@
 # This script helps us only build bazel targets that have been affected by file
 # changes in the branch merging to target branch.
 #
-# We get the list of files from the diff, query for the ones that are part of any bazel target,
-# and then query for the targets that depend on these files.
+# Given a commit range, the scripts writes to stdout all the targets are are affected by the
+# file changes in the commit range.
 #
 # This scripts is inspired by https://github.com/bazelbuild/bazel/blob/master/scripts/ci/ci.sh
 
 set -euo pipefail
 
-set -x
-cd "$(git rev-parse --show-toplevel)"
-
-MERGE_BASE="${MERGE_BASE_SHA:-HEAD}"
-# we can't use HEAD here because that is the merge commit which contains the changes of the current HEAD of master
-COMMIT_RANGE="$MERGE_BASE..${BRANCH_HEAD_SHA:-}"
+COMMIT_RANGE="${1:?Please specify a commit range: "'0deadb33f..HEAD'"}"
+shift
 DIFF_FILES=$(git diff --name-only "${COMMIT_RANGE}")
 
 if grep -qE "(.*\.bazel|.*\.bzl|\.bazelrc|\.bazelversion|mainnet-canister-revisions\.json|^\.github)" <<<"$DIFF_FILES"; then
-    echo "Changes detected in bazel files. Considering all targets." >&2
-    echo ${BAZEL_TARGETS:-"//..."}
+    bazel query //...
```

Comment:
Why not
```suggestion
    echo //...
```

Reply:
because we need the targets to be listed; `//...` wouldn't match the explicit list of targets returned by `bazel query`

Reply:
Oh right because you do the `comm` above.